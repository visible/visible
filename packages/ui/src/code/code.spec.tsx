import { render } from '@testing-library/react';
import React from 'react';

import { Code } from '.';

describe('Code', () => {
  it('matches snapshot', () => {
    const code = `
    function main(params = {}) {
      var json = JSON.stringify(params);
      console.log(json);
    }
    `;

    const { container } = render(<Code language="javascript">{code}</Code>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
