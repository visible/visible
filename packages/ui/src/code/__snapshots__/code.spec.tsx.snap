// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code matches snapshot 1`] = `
<pre
  style="display: block; overflow-x: auto; padding: 0.5em; background: rgb(240, 240, 240); color: rgb(68, 68, 68);"
  theme="[object Object]"
>
  <code>
    
    
    <span
      class="hljs-function"
    >
      <span
        style="font-weight: bold;"
      >
        function
      </span>
       
      <span
        style="color: rgb(136, 0, 0); font-weight: bold;"
      >
        main
      </span>
      (
      <span
        class="hljs-params"
      >
        params = {}
      </span>
      ) 
    </span>
    {
      
    <span
      style="font-weight: bold;"
    >
      var
    </span>
     json = 
    <span
      style="color: rgb(57, 115, 0);"
    >
      JSON
    </span>
    .stringify(params);
      
    <span
      style="color: rgb(57, 115, 0);"
    >
      console
    </span>
    .log(json);
    }
    
  </code>
</pre>
`;
